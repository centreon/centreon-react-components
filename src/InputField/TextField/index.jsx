import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import { makeStyles } from '@material-ui/core/styles';
import MuiTextField from '@material-ui/core/TextField';

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.common.white,
    border: `1px solid ${theme.palette.grey[400]}`,
    borderBottom: 0,
  },
  input: {
    paddingLeft: 8,
    paddingBottom: 4,
  },
  inputHelperText: {
    marginTop: 4,
    textAlign: 'right',
  },
}));

const TextField = forwardRef(function TextField(
  { InputProps, InputLabelProps, FormHelperTextProps, ...other },
  ref,
) {
  const classes = useStyles();

  return (
    <MuiTextField
      InputProps={{
        ...InputProps,
        classes: {
          ...InputProps.classes,
          root: clsx(
            classes.root,
            InputProps.classes && InputProps.classes.root
              ? InputProps.classes.root
              : null,
          ),
          input: clsx(
            classes.input,
            InputProps.classes && InputProps.classes.input
              ? InputProps.classes.input
              : null,
          ),
        },
      }}
      InputLabelProps={{
        ...InputLabelProps,
        shrink: true,
      }}
      FormHelperTextProps={{
        ...FormHelperTextProps,
        classes: {
          root: clsx(
            classes.inputHelperText,
            FormHelperTextProps.classes && FormHelperTextProps.classes.root
              ? FormHelperTextProps.classes.root
              : null,
          ),
        },
      }}
      ref={ref}
      {...other}
    />
  );
});

TextField.propTypes = {
  InputProps: PropTypes.objectOf(PropTypes.any),
  InputLabelProps: PropTypes.objectOf(PropTypes.any),
  FormHelperTextProps: PropTypes.objectOf(PropTypes.any),
};

TextField.defaultProps = {
  InputProps: {},
  InputLabelProps: {},
  FormHelperTextProps: {},
};

export default TextField;
