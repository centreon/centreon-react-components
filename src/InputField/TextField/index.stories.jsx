import React from 'react';

import TextField from '.';

export default { title: 'InputField/Text' };

export const withLabelAndHelperText = () => (
  <TextField label="name" helperText="choose a name for current object" />
);

export const withPlaceholderOnly = () => <TextField placeholder="name" />;

export const withError = () => (
  <TextField error label="name" helperText="Wrong name" />
);

export const fullWidth = () => <TextField fullWidth label="full width" />;
