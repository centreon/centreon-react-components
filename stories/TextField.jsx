import React from 'react';
import { storiesOf } from '@storybook/react';
import TextField from '../src/TextField';

storiesOf('TextField', module).add('with label and helper text', () => (
  <TextField label="name" helperText="choose a name for current object" />
));

storiesOf('TextField', module).add('with placeholder only', () => (
  <TextField placeholder="name" />
));

storiesOf('TextField', module).add('with error', () => (
  <TextField error label="name" helperText="Wrong name" />
));

storiesOf('TextField', module).add('full width', () => (
  <TextField fullWidth label="full width" />
));

storiesOf('TextField', module).add('login / password', () => (
  <div style={{ display: 'flex', flexDirection: 'column', width: 200 }}>
    <TextField label="login" />
    <TextField label="password" type="password" />
  </div>
));
