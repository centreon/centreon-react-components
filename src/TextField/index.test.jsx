import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TextField from '.';

describe('TextField', () => {
  it('renders correctly', () => {
    const { container } = render(
      <TextField
        label="label"
        helperText="helper text"
        placeholder="placeholder"
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('triggers change', async () => {
    const mockOnChange = jest.fn();

    const { getByPlaceholderText } = render(
      <TextField onChange={mockOnChange} placeholder="placeholder" />,
    );

    const input = getByPlaceholderText('placeholder');
    fireEvent.change(input, { target: { value: 'new value' } });

    expect(mockOnChange).toBeCalled();
  });
});
