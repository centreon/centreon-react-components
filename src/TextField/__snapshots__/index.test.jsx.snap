// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders correctly 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
    data-shrink="true"
  >
    label
  </label>
  <div
    class="MuiInputBase-root MuiInput-root makeStyles-root-1 MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input makeStyles-input-2"
      placeholder="placeholder"
      type="text"
      value=""
    />
  </div>
  <p
    class="MuiFormHelperText-root makeStyles-inputHelperText-3"
  >
    helper text
  </p>
</div>
`;
